iModelCore/GeomLibs/serialization/src/FlatBuffer/FixedStructs.cpp

```diff
@@ -2024,20 +2031,27 @@ bvector<IGeometryPtr> *invalidGeometry)
     return false;
     }
 
-
-bool BentleyGeometryFlatBuffer::BytesToPolyfaceQueryCarrier (Byte const *buffer, PolyfaceQueryCarrier &carrier, bool applyValidation)
+bool BentleyGeometryFlatBuffer::BytesToPolyfaceQueryCarrier
+(
+    Byte const *buffer,
+    size_t const bufferSize,
+    PolyfaceQueryCarrier &carrier,
+    bool applyValidation
+)
     {
     if (nullptr == buffer)
         return false;
     Byte const* fbStart = GetFBStart (buffer);
     if (nullptr == fbStart)
-        return false;        
+        return false;
     auto fbRoot = flatbuffers::GetRoot <BGFB::VariantGeometry>(fbStart);
-    if (nullptr != fbRoot
-        && fbRoot->has_geometry_type ()
-        && fbRoot->has_geometry ()
-        )
+    auto myVerifier = flatbuffers::Verifier(fbStart, bufferSize - s_prefixBufferSize);
+    if (!fbRoot->Verify(myVerifier))
+        return false;

```

Comment:
Move the new verifier lines below, inside the if (nullptr != fbRoot && ... ) scope.